storage/src/adb/store/mod.rs

```diff
@@ -184,11 +187,16 @@ pub trait Db<K: Array, V: Codec> {
     fn delete(&mut self, key: K) -> impl Future<Output = Result<bool, Error>>;
 
     /// Commit any pending operations to the database, ensuring their durability upon return from
-    /// this function. Also raises the inactivity floor according to the schedule.
+    /// this function. Also raises the inactivity floor according to the schedule. Returns the
+    /// `(start_loc, end_loc]` location range of committed operations. The end of the returned range
+    /// includes the commit operation itself, and hence will always be equal to `op_count`.
     ///
     /// Failures after commit (but before `sync` or `close`) may still require reprocessing to
     /// recover the database on restart.
-    fn commit(&mut self) -> impl Future<Output = Result<(), Error>>;
+    fn commit(
+        &mut self,
+        metadata: Option<V>,
+    ) -> impl Future<Output = Result<(Location, Location), Error>>;
```

Comment:
I think it might be a bit more Rusty to return `Range<Location>` here. That's also the type we use in the ADB state sync code to represent the range of live operations to be fetched/applied to the log.